src/jep/Jep.java

```diff
@@ -129,7 +132,7 @@ public void run() {
                      * while another thread is in python, then the thread state
                      * can get messed up leading to stability/GIL issues.
                      */
```

Comment:
It is theoretically possible to call call close() here, before the initLock is created. Once you notify TopInterpreter,this there is nothing to prevent that other thread from running long enough to call close() before running this thread any further. I think the simplest solution may be to create initLock earlier and just synchronize the entire method on initLock.